llvm/include/llvm/IR/DataLayout.h

```diff
@@ -564,7 +564,9 @@ inline LLVMTargetDataRef wrap(const DataLayout *P) {
 
 /// Used to lazily calculate structure layout information for a target machine,
 /// based on the DataLayout structure.
-class StructLayout final : public TrailingObjects<StructLayout, TypeSize> {
+class StructLayout final : private TrailingObjects<StructLayout, TypeSize> {
+  friend TrailingObjects;
```

Comment:
For my education, why do we need this? Should't we make the members we need `protected`?

Reply:
See the documentation of TrailingObjects in TrailingObjects.h. Ultimately, we cast "this" to BaseTy in TrailingObjects implementation, and for that to be possible with private inheritance, we need to declare TrailingObjects to be friend (from what I understood).